UnityProject/Assets/Scripts/Learning/ProtipObjectTypes/ProtipObjectOnHealthStateChange.cs

```diff
@@ -94,137 +121,179 @@ private void TriggerConsciousEvent(ConsciousState state)
-					TriggerTip(HungerEvent);
+					TriggerTip(hungerEvent);
 					break;
 				case HungerState.Starving:
-					TriggerTip(StarvingEvent);
+					TriggerTip(starvingEvent);
 					break;
 				default:
 					break;
 			}
+
+			if(lastHungerState != HungerState.Normal) StartCoroutine(HungerCooldown());
 		}
 
 		private void TriggerBleedingTip(BleedingState state)
 		{
+			if(bleedingTipCooldown || IsPlayerGhostInBody()) return;
 			switch (state)
 			{
-				case BleedingState.None:
-					break;
 				case BleedingState.VeryLow:
-					TriggerTip(BleedingVeryLowEvent);
+					TriggerTip(bleedingVeryLowEvent);
 					break;
 				case BleedingState.Low:
-					TriggerTip(BleedingLowEvent);
+					TriggerTip(bleedingLowEvent);
 					break;
 				case BleedingState.Medium:
-					TriggerTip(BleedingHighEvent);
+					TriggerTip(bleedingHighEvent);
 					break;
 				case BleedingState.High:
-					TriggerTip(BleedingHighEvent);
+					TriggerTip(bleedingHighEvent);
 					break;
 				case BleedingState.UhOh:
-					TriggerTip(BleedingDanagerEvent);
+					TriggerTip(bleedingDanagerEvent);
 					break;
 				default:
 					break;
 			}
+			StartCoroutine(BleedingCooldown());
 		}
 
 		private void TriggerFireStacksTip(float state)
 		{
+			if(fireStackTipCooldown || IsPlayerGhostInBody()) return;
 			if (state > 1f)
 			{
-				TriggerTip(FireStacksEvent);
+				TriggerTip(fireStacksEvent);
+				StartCoroutine(FireStackTipCooldown());
 			}
 		}
 
 		private void TriggerSuffocatingTip(bool state)
 		{
-			if (state == false || SufficationTipCooldown) return;
-			TriggerTip(SuffuicationEvent);
+			if (state == false || sufficationTipCooldown || IsPlayerGhostInBody()) return;
+			TriggerTip(suffuicationEvent);
 			StartCoroutine(SufficationCooldown());
 		}
 
 		private void TriggerToxinsTip(bool state)
 		{
-			if(state == false || ToxinTipCooldown) return;
-			TriggerTip(ToxinEvent);
+			if(state == false || toxinTipCooldown || IsPlayerGhostInBody()) return;
+			TriggerTip(toxinEvent);
 			StartCoroutine(ToxinCooldown());
 		}
 
 		private void TriggerTemperatureTip(float state)
 		{
-			if(TempTipOnCooldown) return;
+			if(tempTipOnCooldown || IsPlayerGhostInBody()) return;
 			switch (state)
 			{
 				case <= AtmosConstants.BARELY_COLD_HEAT:
-					TriggerTip(TemperatureColdEvent);
+					TriggerTip(temperatureColdEvent);
 					StartCoroutine(TempCooldown());
 					break;
 				case >= AtmosConstants.HOT_HEAT:
-					TriggerTip(TemperatureHotEvent);
+					TriggerTip(temperatureHotEvent);
 					StartCoroutine(TempCooldown());
 					break;
 				default:
 					break;
 			}
 		}
 
+		private bool IsPlayerGhostInBody()
+		{
+			return healthState.livingHealthMasterBase.playerScript.IsGhost;
+		}
+
 		private IEnumerator TempCooldown()
 		{
-			TempTipOnCooldown = true;
+			tempTipOnCooldown = true;
 			yield return WaitFor.Seconds(PROTIP_COOLDOWN_TO_AVOID_QUEUE_SPAM);
-			TempTipOnCooldown = false;
+			tempTipOnCooldown = false;
 		}
 
 		private IEnumerator PressureCooldown()
```

Comment:
might be better way of doing this generically, as in you passing the ProtipSO into a dictionary<ProtipSO, bool> something like that

Reply:
hm, might do it in another branch. for now this is good.

Reply:
ok as long as it's to do